src/fsharp/PostInferenceChecks.fs

```diff
@@ -741,28 +773,54 @@ and CheckExpr (cenv:cenv) (env:env) origExpr (context:PermitByRefExpr) : bool =
 
     // Allow 'typeof<System.Void>' calls as a special case, the only accepted use of System.Void! 
     | TypeOfExpr g ty when isVoidTy g ty ->
-        false
+        LimitValFlags.None
 
     // Allow 'typedefof<System.Void>' calls as a special case, the only accepted use of System.Void! 
     | TypeDefOfExpr g ty when isVoidTy g ty ->
-        false
+        LimitValFlags.None
 
     // Allow '%expr' in quotations
     | Expr.App(Expr.Val(vref,_,_),_,tinst,[arg],m) when isSpliceOperator g vref && env.quote ->
           CheckTypeInstPermitAllByrefs cenv env m tinst // it's the splice operator, a byref instantiation is allowed
           CheckExprNoByrefs cenv env arg
-          false
+          LimitValFlags.None
 
     // Check an application
     | Expr.App(f,_fty,tyargs,argsl,m) ->
 
         CheckTypeInstNoByrefs cenv env m tyargs
         CheckExprNoByrefs cenv env f
 
+        let contexts = mkArgsForAppliedExpr false argsl f
+
+        let _hasHeapSpanLikeArg, _hasStackSpanLikeArg, hasStackByRefLikeArg =
```

Comment:
This code is a bit complicated, but something like this is necessary when we start supporting stack referring span likes, and mix call functions with arguments that have both stack + heap referring span likes.